compiler/AST/baseAST.cpp

```diff
@@ -187,6 +202,7 @@ void trace_remove(BaseAST* ast, char flag) {
     if (isAlive(ast) || isRootModuleWithType(ast, type)) { \
       g##type##s.v[i##type++] = ast;            \
     } else {                                    \
+      remove_weak_links(ast);                   \
```

Comment:
It would be nice to guard this with `if (E_##type == E_VarSymbol)` and have `remove_weak_links` take a VarSymbol*. Could change this later, though.